[Chong Guang Jun] Duke Increments

tcgj opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

tcgj committed:
Merge branch 'gradle'

tcgj committed:
Add gradle plugins

tcgj committed:
Make duke respond to Level-1 requirements

tcgj committed:
Level up Duke to Level-2

tcgj committed:
Add Task class

tcgj committed:
Level up to Level-3

tcgj committed:
Add support for three types of tasks

tcgj committed:
Level up to Level-4

tcgj committed:
Fix indentation problem for Level-4

tcgj committed:
Add support for A-TextUiTesting

tcgj committed:
append task to task subclass names

tcgj committed:
Add exception class

tcgj committed:
Update gitignore to include test output file

tcgj committed:
Refactor code to support different io streams

tcgj committed:
Fix expected output to include bye message

tcgj committed:
Add exception handling to duke

tcgj committed:
Add delete function

tcgj committed:
Level up to Level-6

tcgj committed:
refactor: refactor duke to better handle exceptions

tcgj committed:
Adjust indentation for switch statement

tcgj committed:
Fix output encoding to utf8

tcgj committed:
Fix bye command not working

tcgj committed:
Set default charset for Duke

tcgj committed:
Add file read and write functionality

tcgj committed:
Give user control of where duke saves its data

tcgj committed:
Level up to Level-7

tcgj committed:
Level up to Level-8

tcgj committed:
Merge branch 'branch-Level-7'

tcgj committed:
Merge branch 'branch-Level-8'

tcgj committed:
Fix load and save not processing date output

tcgj committed:
Package project classes

tcgj committed:
Refactor classes to separate TaskList and UI

tcgj committed:
Refactor codebase to include commands and parser for commands

tcgj committed:
Change file writing to use a bufferedwriter

tcgj committed:
Add standard input to build file

tcgj committed:
Add unit tests for parser and tasklist

tcgj committed:
Rename done boolean to isDone

tcgj committed:
Write javadoc comments for all applicable classes

tcgj committed:
Rename names to meet coding standard

tcgj committed:
Capitalise 'if' for javadoc on throws clause

tcgj committed:
Add new find command

tcgj committed:
Merge branch 'branch-A-JavaDoc'

tcgj committed:
Merge branch 'branch-A-CodingStandard'

tcgj committed:
Merge branch 'branch-Level-9'

tcgj committed:
Update automated testing files

tcgj committed:
Remove dependency on Ui to generate fixed replies for individual commands

tcgj committed:
Re-style some javadoc comments

honhaochen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,104 @@
+                returnCode = executeMainFlow();
+            }
+            storage.save(taskList);
+        } catch (IOException e) {
```
Would it be better if you catch DukeException instead since this is a Duke App?

tcgj committed:
Re-style more javadoc comments

tcgj committed:
Adhere to checkstyle

honhaochen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,106 @@
+     * @param filePath Specifies the path where the <code>TaskList</code> is loaded from.
+     */
+    public Duke(String filePath) {
+        this(new SimpleCliUi(), new Storage(filePath));
```
Would it be better if you just assign <new UI> and <new Storage> in this instance of class Duke? That way you don't need to have two constructors, which might cause some confusion

tcgj replied:
In this case I suppose the intention was so that someone creating a new `Duke` instance could choose to select which `filePath` they wanted to use, if they wanted to specify something rather than the default. Or even more specifically, they could supply the Ui and the Storage themselves.

honhaochen commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,134 @@
+     * @param charset the character encoding to be used.
+     */
+    public Storage(String path, Charset charset) {
+        filePath = Path.of(path).normalize();
```
Could it be just fine with only one constructor?
Pardon me because I personally think with just one will work perfectly fine

tcgj replied:
In this case I'm just handling a case where an implementer might want to use a different encoding than UTF-8 haha

honhaochen commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,134 @@
+     * @return the task list read from file, or a new task list if the file does not exist.
+     * @throws IOException If the file input cannot be read.
+     */
+    public TaskList load() throws IOException {
```
would it be better to throw DukeException instead?

honhaochen commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,134 @@
+                            break;
+                        }
+                    } catch (DukeParserException ignored) {
+                        // If date cannot be parsed, too bad, we shall ignore that task.
```
Hmmm I guess if you can store the task, loading it back should be fine

honhaochen commented on src/main/java/duke/command/CommandBuilder.java:
```diff
@@ -0,0 +1,34 @@
+package duke.command;
```
This file is missing JavaDocs, did you forget about it?

tcgj replied:
Ah, there are no public methods here, so I didn't see a need to add JavaDoc comments here.

honhaochen commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,94 @@
+package duke.command;
```
Parser.java seems to be not part of the commands, might be better if you put into another directory?

honhaochen reviewed (commented):
Comments on Command classes:
- Nice use of abstract classes and inheritance features, it really amazed me in a way, including the CommandBuilder.

Exception Handling:
- I'll suggest that the Duke.java should only deal with Duke exceptions, it should be abstracted away from any java's exceptions (IO Exception for example). One way to deal with it would be catching the exception in the class and throw another Duke-defined exception. It would be useful to deal with multiple types of Exception and would be easier to maintain the code in the future.

Overall:
- Great use of constant declarations to deal with repeated values.
- Great coding conventions, just some JavaDocs are missing.

For details, please look into the specific comments in your codes.
Anyway, great job completing the milestones :)

tcgj committed:
Initial Level-10 commit

tcgj committed:
Merge branch 'branch-Level-10'

aang114 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+    /**
+     * Specifies that <code>Duke</code> should continue to read the next command.
+     */
+    public static final int CODE_CONTINUE = 0;
```
good to see you are following naming covention!

aang114 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+    /**
+     * Specifies the date format used by <code>Duke</code> for receiving date input.
+     */
+    public static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("ddMMyyyy HHmm");
```
maybe it would be better if you made a seperate class to handle converting strings to dates

aang114 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+        this.storage = storage;
+    }
+
+    private int executeMainFlow() throws IOException {
```
good structuring!

aang114 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+        userInput.clear();
+    }
+
+    private String getResponse(String input) {
```
it seems you have not made the gui user interactive yet?

aang114 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,211 @@
+    }
+
+    public static void main(String[] args) {
+        String path = "data/duke.txt";
```
maybe you can make this a final variable?

tcgj committed:
Refactor duke to allow control to be shifted to javafx.

tcgj committed:
Merge branch 'branch-Level-10'

aang114 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,39 @@
+public abstract class Command {
+    protected List<String> arguments;
+
+    Command() {
```
this seems redundant

tcgj replied:
Yes I wish it was also. I needed a package private constructor for the prototype instances of my concrete classes inheriting this. Could do without, but that would not be very good in terms of visibility.

aang114 commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,94 @@
+ * @author Terence Chong Guang Jun
+ */
+public class Parser {
+    private static HashMap<String, CommandBuilder> commandMap;
```
very cool that you are using a HashMap to parse!

aang114 commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,40 @@
+ * @author Terence Chong Guang Jun
+ */
+public class TodoCommand extends AddTaskCommand {
+    TodoCommand() {
```
this seems redundant?

aang114 commented on src/main/java/duke/ui/SimpleCliUi.java:
```diff
@@ -0,0 +1,31 @@
+ *
+ * @author Terence Chong Guang Jun
+ */
+public class SimpleCliUi extends Ui {
```
Wow! Very smart and unique way to represent the command line user interface

aang114 commented on src/test/java/duke/command/ParserTest.java:
```diff
@@ -0,0 +1,30 @@
+
+public class ParserTest {
+    @Test
+    public void parseDate_validDateFormat_success() throws Exception {
```
if i am not wrong, it is not recomended to have methods with '_' (it's against coding standards i think)

tcgj replied:
Haha, perhaps you might wanna look at [https://nus-cs2103-ay1920s1.github.io/website/schedule/week3/topics.html#tools-junit-junit-basic](url)

aang114 reviewed (commented):
Overall your code is very OOP friendly, very easy-to-read and very well implemented - although there were a few places where the coding standard was not followed. Kudos!

tcgj committed:
Merge branch 'branch-A-CheckStyle'

tcgj committed:
Change to adhere to checkstyle.

tcgj committed:
Add assertion usage

tcgj committed:
Refactor code to improve code quality (#6)

* Remove redundant imports

* Refactor parser to make happy path more prominent

* Refactor storage to make happy path more prominent

tcgj committed:
Add assertion for reading tasks from file

tcgj committed:
Add support for friendlier syntax (#8)

* Restructure commands to use method reference instead of prototypes

* Create command aliases

* Change accessibility of command classes

* Add alias command to allow user to create own aliases

tcgj committed:
Set theme jekyll-theme-slate

tcgj committed:
Set theme jekyll-theme-cayman

tcgj committed:
Set theme jekyll-theme-midnight

tcgj committed:
Set theme jekyll-theme-hacker

tcgj committed:
Set theme jekyll-theme-cayman

tcgj committed:
Set theme jekyll-theme-hacker

tcgj committed:
Clean up

tcgj committed:
Update gh pages

tcgj committed:
Merge branch 'master' of https://github.com/tcgj/duke

tcgj committed:
Add horizontal rules

tcgj committed:
Update gradle build file

tcgj committed:
Update gradle build file to support cross-platform

tcgj committed:
Refactor code

- Change all references from duke to daming
- Add persistence for aliases through savefile

tcgj committed:
Change unicode characters to use escaped sequence

tcgj committed:
Remove redundant dependencies

tcgj committed:
Make minor adjustments to javafx gui

tcgj committed:
Make more adjustments to javafx gui
